### PR TITLE
Add documentation of configuration schema

### PR DESCRIPTION
**Pull request by mimi1vx**

### CODE REVIEW

File: doc/config.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Configuration format of qem-bot
+
+Configuration,in another way also called metadata of qem-bot is hold in yaml files with one file per "product". Plus there is one special config file called `singlearch.yml` which contains list of packages that exist only on single architecture.
```
Comment (Martchus) [inline doc/config.md:3]:
```suggestion
Configuration, in another way also called metadata of qem-bot, is hold in yaml files with one file per "product". Additionally, there is one special config file called `singlearch.yml` which contains a list of packages that exist only on single architecture.
```

File: doc/config.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Example of `singlearch.yaml`:
+
+```yaml
+- packageone
+- packagetwo
+- onlyppcle
+```
+
+If incident has package from this list, job is automaticaly marked as without **aggregate**. With this incident can be approved without existing aggregate jobs.
```
Comment (Martchus) [inline doc/config.md:13]:
```suggestion
If an incident has a package from this list, a job is automatically marked as without **aggregate**. That means the incident can be approved without existing aggregate jobs.
```

File: doc/config.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+Rest yaml files contains definition of **product** and data either or both **aggregate** and **incidents**
```
Comment (Martchus) [inline doc/config.md:14]:
```suggestion
The other yaml files contain the definition of **product** and data either or both **aggregate** and **incidents**
```
or
```suggestion
The other yaml files contain definitions of **product** and data either or both **aggregate** and **incidents**
```
or something else? I don't really understand this sentence. Especially the last part.

File: doc/config.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+## Structure of product definition
```
Comment (Martchus) [inline doc/config.md:16]:
```suggestion
## Structure of a product definition
```

File: doc/config.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+product: leap155
+settings:
+  DISTRI: opensuse
+  VERSION: '15.5'
+  SOMETHING: 'something123'
+  ELSE: 'ELSE123'
+```
+
+* `product` - mandatory and must be unique, no other config file can have same product. 
+* `settings` - mandatory dictionary containing `key` : `value` which are used directly in job schedule. From this pairs are `DISTRI` and `VERSION` mandatory.
```
Comment (Martchus) [inline doc/config.md:30]:
```suggestion
* `settings` - mandatory dictionary containing `key` : `value` which are used directly in job schedule. From this the pairs `DISTRI` and `VERSION` are mandatory.
```

File: doc/config.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+## Aggregate part of config
+
+Used for schedule multiple Incidents in product as one BUILD named **YYYYMMDD-{counter from 1}**. Jobs scheduled on this part of config will have multiple vars `*_TEST_REPO`  used add Incidents to test.
```
Comment (Martchus) [inline doc/config.md:34]:
```suggestion
Used to schedule multiple Incidents in a product as one BUILD named **YYYYMMDD-{counter from 1}**. Jobs scheduled on this part of the config will have multiple vars `*_TEST_REPO`  used to add Incidents to test. 
```

File: doc/config.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    BASE_TEST_ISSUES: Basesystem:17-SP5
+    OS_TEST_ISSUES: Product-Linux:17-SP5
+    SERVERAPP_TEST_ISSUES: Module-Server-Applications:17-SP5
+    DESKTOP_TEST_ISSUES: Module-Desktop-Applications:17-SP5
+    SDK_TEST_ISSUES: Development-Tools:17-SP5
+    PCM_TEST_ISSUES: Public-Cloud:17-SP5
+  onetime: true
+```
+
+* `aggregate` - self explanatory, mapping containing all needed settings for aggregate schedule of given *product*
```
Comment (Martchus) [inline doc/config.md:54]:
```suggestion
* `aggregate` - self-explanatory mapping containing all needed settings for the aggregate schedule of a given *product*
```

File: doc/config.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+* `FLAVOR` - flavor used by schedule, can be only one per product, mandatory
```
Comment (Martchus) [inline doc/config.md:55]:
```suggestion
* `FLAVOR` - flavor used by the schedule, can be only one per product, mandatory
```

File: doc/config.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+* `archs` - list scheduled architectures, mandatory
+* `test_issues` - mandatory, contains mapping of `*_TEST_ISSUES`, which are used to decide which incidents to be scheduled in aggregate jobs. Values are `PRODUCT-IDENTIFICATION:VERSION`. `PRODUCT_IDENTIFICATION` as is defined in OBS/IBS, `VERSION` dtto, could use different versions for different `*_TEST_ISSUES`.  `OS_TEST_ISSUES` - implicit, always used by *os-autoinst-distri-opensuse*, contains identification of base product. All other contains modules, addons, extensions used in this aggreggate. First part of key name must be same (uppercase, os-autoinst-distri-opensuse wll convert it to lowercase) as addon,extension,module identification in `SCC_ADDONS` variable defined in job template inside openqa instance.
```
Comment (Martchus) [inline doc/config.md:57]:
```suggestion
* `test_issues` - mandatory, contains mapping of `*_TEST_ISSUES`, which are used to decide which incidents to be scheduled in aggregate jobs. Values are `PRODUCT-IDENTIFICATION:VERSION`. `PRODUCT_IDENTIFICATION` as is defined similarly in OBS/IBS, `VERSION`, could use different versions for different `*_TEST_ISSUES`.  `OS_TEST_ISSUES` - implicit, always used by *os-autoinst-distri-opensuse*, contains identification of base product. All others contain modules, add-ons, extensions used in this aggregate. First part of key name must be same (uppercase, os-autoinst-distri-opensuse will convert it to lowercase) as add-on,extension,module identification in `SCC_ADDONS` variable defined in job template inside openQA instance.
```

File: doc/config.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+* `onetime` - optional key, boolean.  By default qem-bot implies False. When used with True value it limits bot scheduling this aggregate only once per day.
```
Comment (Martchus) [inline doc/config.md:58]:
```suggestion
* `onetime` - optional key, boolean.  By default value is False. When set to True, it limits scheduling this aggregate only once per day.
```

File: doc/config.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+        - pkgfour
+    Other-Incident-FLavor:
+      ...
+```
+
+* `incidents` mapping contains all settings for incident schedule of product.
+* `FLAVOR` mapping containing FLAVOR's and all settings needed for FLAVOR schedule. FLAVOR itself maps 1:1 to openQA's FLAVOR
+* `archs` - list of scheduled architectures, mandatory
+* `issues` - mandatory, same rules as `test_issues` of aggregate
+* `required_issues` - optional, contains list of  `issues` keys which must be part of Incident to be scheduled.  Without this key any incident containing any target will be scheduled in this FLAVOR.
```
Comment (Martchus) [inline doc/config.md:107]:
```suggestion
* `required_issues` - optional, contains list of  `issues` keys which must be part of Incident to be scheduled. Without this key any incident containing any target will be scheduled in this FLAVOR.
```

File: doc/config.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+* `aggregate_job` - optional, boolean value, by default is `True`. When set to `False` it indicates this FLAVOR don't need aggregate job for approval.
+* `aggregate_check_true` and `aggregate_check_false` - optional, list of keywords, used only with `aggregate_job: false`. Conditions finely tunning need of Aggregate for approval. Keywords are settings keys scheduled for job.
```
Comment (Martchus) [inline doc/config.md:109]:
```suggestion
* `aggregate_check_true` and `aggregate_check_false` - optional, list of keywords, used only with `aggregate_job: false`. Conditions for finely tuning the need of Aggregate for approval. Keywords are settings keys scheduled for the job. 
```

File: doc/config.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+* `incidents` mapping contains all settings for incident schedule of product.
+* `FLAVOR` mapping containing FLAVOR's and all settings needed for FLAVOR schedule. FLAVOR itself maps 1:1 to openQA's FLAVOR
+* `archs` - list of scheduled architectures, mandatory
+* `issues` - mandatory, same rules as `test_issues` of aggregate
+* `required_issues` - optional, contains list of  `issues` keys which must be part of Incident to be scheduled.  Without this key any incident containing any target will be scheduled in this FLAVOR.
+* `aggregate_job` - optional, boolean value, by default is `True`. When set to `False` it indicates this FLAVOR don't need aggregate job for approval.
+* `aggregate_check_true` and `aggregate_check_false` - optional, list of keywords, used only with `aggregate_job: false`. Conditions finely tunning need of Aggregate for approval. Keywords are settings keys scheduled for job. 
+* `override_priority` - optional, interger. Overrides default priority of job. (default = 50, with modifiers for `Minimal`, EMU and staging Incidents)
```
Comment (Martchus) [inline doc/config.md:110]:
```suggestion
* `override_priority` - optional, interger. Overrides the default priority of the job. (default = 50, with modifiers for `Minimal`, EMU and staging Incidents)
```

File: doc/config.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+* `packages` - optional, list of package names (or first part of pkg name). Incident must contain package from this list to be scheduled into openQA
```
Comment (Martchus) [inline doc/config.md:111]:
```suggestion
* `packages` - optional, list of package names (or first part of package name). An Incident must contain a package from this list to be scheduled into openQA
```

File: doc/config.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+* `exluded_packages` - optional, list of pacage names, opposite to `packages`
```
Comment (Martchus) [inline doc/config.md:112]:
```suggestion
* `excluded_packages` - optional, list of package names, opposite to `packages`
```

**Review by Martchus — CHANGES_REQUESTED** [review]
The lack of articles/prepositions any many places makes it hard to follow. So don't insert my suggestions blindly because I might not have been able to guess the meaning correctly.

File: doc/config.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+Example of `singlearch.yaml`:
+
+```yaml
+- packageone
+- packagetwo
+- onlyppcle
+```
+
+If a incident has package from this list, a job is automaticaly marked as without **aggregate**. That means the Incident can be approved without existing aggregate jobs.
```
Comment (baierjan) [inline doc/config.md:13]:
```suggestion
If an Incident has package from this list, a job is automatically marked as without **aggregate**. That means the Incident can be approved without existing aggregate jobs.
```

File: doc/config.md [hunk 17]
```diff
@@ -0,0 +1,116 @@
+# Configuration format of qem-bot
+
+Configuration, in another way also called metadata of qem-bot is hold in yaml files with one file per "product". Additionaly, there is one special config file called `singlearch.yml` which contains a list of packages that exist only on single architecture.
```
Comment (baierjan) [inline doc/config.md:3]:
```suggestion
Configuration, in another way also called metadata of qem-bot is hold in yaml files with one file per **product**. Additionaly, there is one special config file called `singlearch.yml` which contains a list of packages that exist only on single architecture.
```

File: doc/config.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+- packagetwo
+- onlyppcle
+```
+
+If a incident has package from this list, a job is automaticaly marked as without **aggregate**. That means the Incident can be approved without existing aggregate jobs.
+Other yaml files contain the definition of **product** and data either or both **aggregate** and **incidents**
+
+## Structure of a product definition
+
+Contains unique identification of a product, common settings and `DISTRI` plus `VERSION` variables for openQA.
```
Comment (baierjan) [inline doc/config.md:18]:
```suggestion
Contains unique identification of a **product**, common settings and `DISTRI` plus `VERSION` variables for openQA.
```

File: doc/config.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+    OS_TEST_ISSUES: Product-Linux:17-SP5
+    SERVERAPP_TEST_ISSUES: Module-Server-Applications:17-SP5
+    DESKTOP_TEST_ISSUES: Module-Desktop-Applications:17-SP5
+    SDK_TEST_ISSUES: Development-Tools:17-SP5
+    PCM_TEST_ISSUES: Public-Cloud:17-SP5
+  onetime: true
+```
+
+* `aggregate` - self-explanatory, mapping containing all needed settings for the aggregate schedule of a given *product*
+* `FLAVOR` - flavor used by the schedule, can be only once per product, mandatory
```
Comment (baierjan) [inline doc/config.md:57]:
```suggestion
* `FLAVOR` - flavor used by the schedule, can be only once per **product**, mandatory
```

File: doc/config.md [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+* `archs` - list scheduled architectures, mandatory
+* `test_issues` - mandatory, contains mapping of `*_TEST_ISSUES`, which are used to decide which incidents to be scheduled in aggregate jobs. Values are `PRODUCT-IDENTIFICATION:VERSION`. `PRODUCT_IDENTIFICATION` as is defined in OBS/IBS. `VERSION` dtto, could use different versions for different `*_TEST_ISSUES`.  `OS_TEST_ISSUES` - implicit, always used by *os-autoinst-distri-opensuse*, contain identification of the base product. All others contain modules, addons, extensions used in this aggregate. First part of key name must be same (uppercase, os-autoinst-distri-opensuse will convert it to lowercase) as addon, extension or module identification in `SCC_ADDONS` variable defined in job template inside openQA instance.
```
Comment (baierjan) [inline doc/config.md:59]:
```suggestion
* `test_issues` - mandatory, contains mapping of `*_TEST_ISSUES`, which are used to decide which Incidents to be scheduled in aggregate jobs. Values are `PRODUCT-IDENTIFICATION:VERSION`. `PRODUCT-IDENTIFICATION` as is defined in OBS/IBS. `VERSION` dtto, could use different versions for different `*_TEST_ISSUES`. 
* `OS_TEST_ISSUES` - implicit, always used by *os-autoinst-distri-opensuse*, contain identification of the base product. All others contain modules, addons, extensions used in this aggregate. First part of key name must be same (uppercase, *os-autoinst-distri-opensuse* will convert it to lowercase) as addon, extension or module identification in `SCC_ADDONS` variable defined in job template inside openQA instance.
```

File: doc/config.md [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+      ...
+```
+
+* `incidents` mapping contains all settings for incident schedule of a product.
+* `FLAVOR` mapping containing FLAVORs and all settings needed for to schedule them. FLAVOR itself maps 1:1 to an openQA's FLAVOR
+* `archs` - list of scheduled architectures, mandatory
+* `issues` - mandatory, same rules as `test_issues` of aggregate
+* `required_issues` - optional, contains list of `issues` keys which must be part of the Incident to be scheduled. Without this key any incident containing any target in `issues` will be scheduled in this FLAVOR.
+* `aggregate_job` - optional, boolean value, by default is `True`. When set to `False` it indicates this FLAVOR doesn't need an aggregate job for approval.
+* `aggregate_check_true` and `aggregate_check_false` - optional, list of keywords, used only with `aggregate_job: false`. Conditions for finely tunning the need of the Aggregate jobs for approval. Keywords are settings keys scheduled used in job.
```
Comment (baierjan) [inline doc/config.md:111]:
```suggestion
* `aggregate_check_true` and `aggregate_check_false` - optional, list of keywords, used only with `aggregate_job: false`. Conditions for finely tuning the need of the aggregate jobs for approval. Keywords are settings keys scheduled used in job. 
```

File: doc/config.md [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+* `incidents` mapping contains all settings for incident schedule of a product.
+* `FLAVOR` mapping containing FLAVORs and all settings needed for to schedule them. FLAVOR itself maps 1:1 to an openQA's FLAVOR
+* `archs` - list of scheduled architectures, mandatory
+* `issues` - mandatory, same rules as `test_issues` of aggregate
+* `required_issues` - optional, contains list of `issues` keys which must be part of the Incident to be scheduled. Without this key any incident containing any target in `issues` will be scheduled in this FLAVOR.
+* `aggregate_job` - optional, boolean value, by default is `True`. When set to `False` it indicates this FLAVOR doesn't need an aggregate job for approval.
+* `aggregate_check_true` and `aggregate_check_false` - optional, list of keywords, used only with `aggregate_job: false`. Conditions for finely tunning the need of the Aggregate jobs for approval. Keywords are settings keys scheduled used in job. 
+* `override_priority` - optional, interger. Overrides the default priority of the job. (default = 50, with modifiers for `*Minimal`, EMU and staging Incidents)
```
Comment (baierjan) [inline doc/config.md:112]:
```suggestion
* `override_priority` - optional, integer. Overrides the default priority of the job (default = 50, with modifiers for `*Minimal`, EMU and staging Incidents).
```

File: doc/config.md [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+* `packages` - optional, list of package names (or first part of pkg name). Incident must contain package from this list to be scheduled into openQA
+* `excluded_packages` - optional, list of package names, opposite to `packages`. If the Incident contains package in this list, isn't scheduled.
```
Comment (baierjan) [inline doc/config.md:114]:
```suggestion
* `excluded_packages` - optional, list of package names, opposite to `packages`. If the Incident contains package in this list, it isn't scheduled.
```

**Review by baierjan — APPROVED** [review]
Just some nit-picks in the sake of consistency, otherwise it feel fine.

File: doc/config.md [hunk 24]
```diff
@@ -0,0 +1,118 @@
+# Configuration format of qem-bot
+
+Configuration, in another way also called metadata of qem-bot is held in yaml files with one file per "product". Additionally, there is one special config file called `singlearch.yml` which contains a list of packages that exist only on a single architecture.
```
Comment (kraih) [inline doc/config.md:3]: Since the configuration spans multiple files, maybe it would be a good idea to start this with a directory listing of these YAML files, before describing them individually in more detail. To get an overview of what's being described in the document.
```
|- singlearch.yaml
...
```

**Comment by okurz** [comment]
+1

File: doc/config.md [hunk 25]
```diff
@@ -0,0 +1,129 @@
...
+```bash
+/etc/openqabot ❯❯❯ tree                   
+.                          
+├── singlearch.yml               
+├── leap154.yml                   
+├── sles15sp3.yml        
+├── micro53.yml          
+...
+```
+Example of `singlearch.yaml`:
```
Comment (kraih) [inline doc/config.md:15]: This says `.yaml` and the two mentions above say `.yml`. Do both extensions work? Then it's probably worth mentioning that as well.

File: doc/config.md [hunk 26]
```diff
@@ -0,0 +1,129 @@
+# Configuration format of qem-bot
+
+Configuration, in another way also called metadata of qem-bot, is held in yaml files with one file per "product". Additionally, there is one special config file called `singlearch.yml` which contains a list of packages that exist only on a single architecture.
+Configuration usually spans over multiple files. And default location excepted from qem-bot is `/etc/openqabot/`
```
Comment (perlpunk) [inline doc/config.md:4]:
```suggestion
Configuration usually spans over multiple files. The default location accepted from qem-bot is `/etc/openqabot/`.
```

File: doc/config.md [hunk 27]
```diff
@@ -0,0 +1,129 @@
...
+    OS_TEST_ISSUES: Product-Linux:17-SP5
+    SERVERAPP_TEST_ISSUES: Module-Server-Applications:17-SP5
+    DESKTOP_TEST_ISSUES: Module-Desktop-Applications:17-SP5
+    SDK_TEST_ISSUES: Development-Tools:17-SP5
+    PCM_TEST_ISSUES: Public-Cloud:17-SP5
+  onetime: true
+```
+
+* `aggregate` - self-explanatory, mapping containing all needed settings for the aggregate schedule of a given *product*
+* `FLAVOR` - flavour used by the schedule, can be only once per product, mandatory
```
Comment (perlpunk) [inline doc/config.md:68]: I would add the `mandatory` either at the beginning or at the end of the descriptions, but consistently

File: doc/config.md [hunk 28]
```diff
@@ -0,0 +1,129 @@
...
+* `archs` - list scheduled architectures, mandatory
+* `test_issues` - mandatory, contains a mapping of `*_TEST_ISSUES`, which are used to decide which Incidents to be scheduled in aggregate jobs. Values are `PRODUCT-IDENTIFICATION:VERSION`. `PRODUCT-IDENTIFICATION` as defined in OBS/IBS. `VERSION` dtto, could use different versions for different `*_TEST_ISSUES`.
```
Comment (perlpunk) [inline doc/config.md:70]: "dtto"?

File: doc/config.md [hunk 29]
```diff
@@ -0,0 +1,129 @@
...
+  * `OS_TEST_ISSUES` variable is implicit, always used by *os-autoinst-distri-opensuse*, contain identification of the base product.
```
Comment (perlpunk) [inline doc/config.md:71]:
```suggestion
  * `OS_TEST_ISSUES` variable is implicit, always used by *os-autoinst-distri-opensuse*, contains identification of the base product.
```

File: doc/config.md [hunk 30]
```diff
@@ -0,0 +1,129 @@
...
+        - pkgone
+        - pkgtwo
+      excluded_packages:
+        - pkgthree
+        - pkgfour
+    Other-Incident-FLavor:
+      ...
+```
+
+* `incidents` mapping contains all settings for an incident schedule of a product.
```
Comment (perlpunk) [inline doc/config.md:118]:
```suggestion
* `incidents` mapping containing all settings for an incident schedule of a product.
```

File: doc/config.md [hunk 31]
```diff
@@ -0,0 +1,129 @@
...
+* `FLAVOR` mapping containing FLAVORs and all settings needed to schedule them. FLAVOR itself maps 1:1 to an openQA's FLAVOR
+* `archs` - list of scheduled architectures, mandatory
+* `issues` - mandatory, same rules as `test_issues` of aggregate
+* `required_issues` - optional, contains a list of `issues` keys which must be part of the Incident to be scheduled. Without this key, any incident containing any target in `issues` will be scheduled in this FLAVOR.
+* `aggregate_job` - optional, boolean value, by default is `True`. When set to `False` it indicates this FLAVOR doesn't need an aggregate job for approval.
```
Comment (perlpunk) [inline doc/config.md:123]:
```suggestion
* `aggregate_job` - optional, boolean value, by default is `true`. When set to `false` it indicates this FLAVOR doesn't need an aggregate job for approval.
```
We should use the canonical YAML boolean values

File: doc/config.md [hunk 32]
```diff
@@ -0,0 +1,129 @@
...
+* `aggregate_check_true` and `aggregate_check_false` - optional, list of keywords, used only with `aggregate_job: false`. Conditions for finely tuning the need of the Aggregate jobs for approval. Keywords are settings keys scheduled used in a job.
```
Comment (perlpunk) [inline doc/config.md:124]:
```suggestion
* `aggregate_check_true` and `aggregate_check_false` - optional, list of keywords, used only with `aggregate_job: false`. Conditions for fine tuning the need of the Aggregate jobs for approval. Keywords are settings keys scheduled used in a job.
```
And I don't understand the last sentence.

**Review by perlpunk — DISMISSED** [review]

File: doc/config.md (hunk 28 above)
Comment (Martchus) [reply doc/config.md:70]: Likely "ditto" - which means "I'm bragging with my knowledge of antiquated Latin words". So my suggestion was to use "similarly" instead.

File: doc/config.md (hunk 31 above)
Comment (mimi1vx) [reply doc/config.md:123]: really ? for YAML 1.1  - is canonical YAML boolean value `y` or `n` ..and parserve we using -> PyYAML is YAML 1.1. 

https://yaml.org/type/bool.html

**Review by baierjan — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

File: doc/config.md (hunk 25 above)
Comment (kraih) [reply doc/config.md:15]: This was marked as resolved, but i don't see anything in the docs about it.

File: doc/config.md (hunk 31 above)
Comment (perlpunk) [reply doc/config.md]: Ah right, it is `y|n` for YAML 1.1, but it is a good practice to use `true|false` like in YAML 1.2 already.

File: doc/config.md (hunk 26 above)
Comment (perlpunk) [reply doc/config.md:4]: This wasn't addressed

File: doc/config.md (hunk 27 above)
Comment (perlpunk) [reply doc/config.md:68]: This wasn't addressed.
What I mean is: for all items, either put the "mandatory" at the beginning of the description, or at the end. Same with optional, I would say.
It can be a bit confusing to have it sometimes at the start and sometimes at the end.

File: doc/config.md (hunk 30 above)
Comment (perlpunk) [reply doc/config.md:118]: This wasn't addressed.
Here you use `mapping contains`, and directly below you use `mapping containing`.

File: doc/config.md (hunk 25 above)
Comment (mimi1vx) [reply doc/config.md:15]: ahh, forgot commit. Thx for spotting
